lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3464,4 +3465,41 @@ public function hydrationComplete()
     {
         $this->hydrationCompleteHandler->hydrationComplete();
     }
+
+    /**
+     * Finds newly persisted entities in entityInsertions by a set of criteria
+     *
+     * @param string     $entityName
+     * @param array      $criteria
+     * @param array|null $orderBy
+     *
+     * @return array
+     */
+    public function findInEntityInsertions($entityName, array $criteria, array $orderBy = null)
+    {
+        $result = array();
```

Comment:
First thing to do here is to check whether the criteria contains an identifier. If not: bail out